cilium-sysdump/__main__.py

```diff
@@ -24,12 +26,31 @@
 import time
 import distutils.util
 
+
 log = logging.getLogger(__name__)
 
 
+def is_ip_or_name(s):
+    if not s:
+        return False
+
+    # We first check if we have any indications that the string is an IPv4 or
+    # an IPv6 address. If so, we'll try to convert the string to an IP address.
+    # If that throws, then we know it's invalid and we'll warn the user.
+    if '.' in s or ':' in s:
+        if not net.is_ipaddress(s):
+            raise argparse.ArgumentTypeError('Invalid IP address')
```

Comment:
Just one concern here: node names may contain a `.`. Such node names would be impossible to specify by the user with the current implementation. I wonder if we shouldn't just assume that if we can't read `s` as an IP address, then it's likely a node name (at least, until you implement the TODO down below).

Reply:
Synced offline. Decided to remove the validation logic and simply assume that if the user enters an IP and it is not an IP, then we assume it's a node name. Also, introduced a log msg to indicate to the user what we are filtering on.